### DISCUSSION TITLE
expr functions with alias

### DISCUSSION THREAD

**Discussion by kennethsato** [mikro-orm/mikro-orm#3535] [Q&A] [answered]
We have the following find:

```
const comerciais = await this.ormRepository.find(
      {
        midia: {
          avaliacaoeng: 'reprovado',
          [expr('date(checked_at)')]: data,
        },
      },
        populate: [
          'midia',
        ],
       fields: (...)
      },
```
And this is the resulting query:
select e0.id, e0.titulo, e0.duracao, e0.codmaterial, e0.ptovenda, e0.created_at, e0.primeiraveiculacao, e0.cliente_id, e0.agencia_id, e0.produtora_id, e0.midia_id from comercial as e0 left join midia as e1 on e0.midia_id = e1.id where e1.avaliacaoeng = 'reprovado' **and e1.date(checked_at) = '2022-04-04'**

We expected the last filter to be  date( **e1.** checked_at) = '2022-04-04' instead of **e1**.date(checked_at) = '2022-04-04'

Is there any way to put the alias inside our SQL function?

**Comment by B4nan** [comment, marked as answer]
``expr(alias => `date(${alias}.checked_at)`)``

**Reply by kennethsato** [reply]
Hello @B4nan .

I'm not sure if I get it.

Tried like this:
```
    const comerciais = await this.ormRepository.find(
      {
        midia: {
          avaliacaoeng: 'reprovado',
          [expr(alias => `date(${alias}.checked_at)`)]: data,
        },
      },
      {
        fields: (...),
        populate: [
          'midia',
        ],
      },
    );
```
It did not work.
Getting this TS error:
"Argument of type '(alias: any) => string' is not assignable to parameter of type 'string'"
And MySQL syntax error:
previous Error: select e0.id, e0.titulo, e0.duracao, e0.codmaterial, e0.ptovenda, e0.created_at, e0.primeiraveiculacao, e0.cliente_id, e0.agencia_id, e0.produtora_id, e0.midia_id from comercialas e0 left join midia as e1 on e0.midia_id = e1.id where e1.avaliacaoeng = 'reprovado' and e1.alias => date(${alias}.checked_at) = '2022-09-01' - You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '=> date(${alias}.checked_at) = '2022-09-01'' at line 1

**Reply by B4nan** [reply]
What version are you on?

**Reply by kennethsato** [reply]
Oh.... So sorry
Outdated again...

I was on 4.5.9.
I've updated to 5.4.2 (also had to update NodeJS) and now it works.

Thanks a lot @B4nan

**Comment by kennethsato** [comment]
Hello @B4nan 

The alias solution is working just fine.
But I tried to remove the alias, and I got the following error:

> "select `c0`.`id`, `c0`.`titulo`, `c0`.`duracao`, `c0`.`codmaterial`, `c0`.
> `categoria`, `c0`.`ptovenda`, `c0`.`created_at`, `c0`.`primeiraveiculacao`, `c0`.`clie
> nte_id`, `c0`.`agencia_id`, `c0`.`produtora_id`, `c0`.`responsavelopec_id`, `c0`.`midi
> a_id`, `c1`.`id` as `c1__id`, `c1`.`mnemonico` as `c1__mnemonico`, `a2`.`id` as `a2__i
> d`, `a2`.`mnemonico` as `a2__mnemonico`, `p3`.`id` as `p3__id`, `p3`.`nome` as `p3__no
> me`, `r4`.`id` as `r4__id`, `r4`.`login` as `r4__login`, `r4`.`nome` as `r4__nome`, `m
> 5`.`id` as `m5__id`, `m5`.`categoria` as `m5__categoria`, `m5`.`obseng` as `m5__obseng
> `, `m5`.`statuspreavaliacao` as `m5__statuspreavaliacao`, `m5`.`proxypath` as `m5__pro
> xypath`, `m5`.`md5` as `m5__md5`, `m5`.`provided_at` as `m5__provided_at`, `m5`.`check
> ed_at` as `m5__checked_at`, `m5`.`avaliador_id` as `m5__avaliador_id`, `a6`.`id` as `a
> 6__id`, `a6`.`login` as `a6__login`, `a6`.`nome` as `a6__nome`, `a6`.`avatar` as `a6__
> avatar` from `comercial` as `c0` left join `cliente` as `c1` on `c0`.`cliente_id` = `c
> 1`.`id` left join `agencia` as `a2` on `c0`.`agencia_id` = `a2`.`id` left join `produt
> ora` as `p3` on `c0`.`produtora_id` = `p3`.`id` left join `usuario` as `r4` on `c0`.`r
> esponsavelopec_id` = `r4`.`id` left join `midia` as `m5` on `c0`.`midia_id` = `m5`.`id
> ` left join `usuario` as `a6` on `m5`.`avaliador_id` = `a6`.`id` left join `midia` as
> `m7` on `c0`.`midia_id` = `m7`.`id` where `c0`.`deleted_at` is null and date(checked_a
> t) = '2025-07-25' and `m7`.`avaliacaoeng` = 'reprovado' and `m7`.`categoria` != 'patro
> cinio' and `c0`.`categoria` != 'patrocinio' order by `m7`.`checked_at` desc, `m5`.`che
> cked_at` desc - Column 'checked_at' in where clause is ambiguous",

And it got me wondering: why is it creating a duplicated left join with midia (m5 and m7)?

**Reply by B4nan** [reply]
One is used for loading the relation, the other for filtering the root entity - by default, populated relations wont respect your where query, you can control this via `populateWhere` option, setting it to `infer` will use the same join branch (but populate less results if the where filter disqualifies some rows), the default is `all`.

**Reply by kennethsato** [reply]
Oh got it! Perfect!

Thanks!
